pytorch_lightning/utilities/fetching.py

```diff
@@ -158,13 +154,13 @@ def loader_iters(self) -> List[Iterator]:
 
     @property
     def state(self) -> Any:
-        def collect_state(iterator: Iterator):
+        def collect_state(iterator: Iterator) -> Any:
```

Comment:
I'm not sure about the correct return type here: Would it be the one of `MergedIteratorState.state`?

Reply:
I believe it should be Collection of `MergedIteratorState`